pep-3156.txt

```diff
@@ -1647,15 +1720,31 @@ callback-based frameworks like Twisted.  After converting a coroutine
 into a Task, callbacks can be added to the Task.
 
 To convert a coroutine into a task, call the coroutine function and
-pass the resulting coroutine object to the ``asyncio.Task()``
-constructor.  You may also use ``asyncio.async()`` for this purpose.
+pass the resulting coroutine object to the ``loop.create_task()``
+method.  You may also use ``asyncio.ensure_future()`` for this purpose.
 
 You may ask, why not automatically convert all coroutines to Tasks?
 The ``@asyncio.coroutine`` decorator could do this.  However, this would
 slow things down considerably in the case where one coroutine calls
 another (and so on), as switching to a "bare" coroutine has much less
 overhead than switching to a Task.
 
+A Task is inherited from ``Future`` but adds new methods:
```

Comment:
They share quite a few APIs (e.g. result(), cancel()) and behaviors (being `await`-able). The big difference is that if you're creating Futures you're responsible for calling set_result()/set_exception() (presumably in some callback), while if you're creating Tasks that will be done for you (based on the lifecycle of the coroutine used to create it). It doesn't seem a big burden to learn and remember that difference, but I wish I could interview the people who made that mistake. (I suspect they first learned callback-based programming, perhaps for a different framework.)

There's a trick possible if we want to catch people who mistakenly call set_result() on a Task, e.g. from inside their coroutine. The task runner can call Future.set_result() (or perhaps super().set_result()), and then Task can override set_result() to just raise an error. Ditto for set_exception().

(Note that cancel() vs. set_result() is a different can of worms, indeed people too often believe that cancel() is an alternative way of setting the result, while it is actually an alternative way of *consuming* the result -- it means the intended consumer has lost interest.)